src/pudl_rmi/coordinate.py

```diff
@@ -79,16 +89,60 @@ def plant_parts_eia(self, clobber=False):
             )
             # actually make the master plant parts list
             plant_parts_eia = self.pudl_out.plant_parts_eia()
+            # verify that record_id_eia is the index
+            if plant_parts_eia.index.name != "record_id_eia":
+                logger.error("Plant parts list index is not record_id_eia.")
             plant_parts_eia = plant_parts_eia[
                 ~plant_parts_eia.index.duplicated(keep="first")
             ]
+            # temporary: make the object columns strings to reduce memory
+            # will be moved to PUDL repo
+            plant_parts_eia = _set_plant_part_dtypes(plant_parts_eia)
             # export
             plant_parts_eia.to_pickle(file_path)
         else:
             logger.info(f"Reading the EIA plant-parts from {file_path}")
             plant_parts_eia = pd.read_pickle(file_path)
+            if plant_parts_eia.index.name != "record_id_eia":
+                logger.error("Plant parts list index is not record_id_eia.")
+            plant_parts_eia = _set_plant_part_dtypes(plant_parts_eia)
+
+        if pickle_distinct:
+            distinct_file_path = pudl_rmi.DISTINCT_PLANT_PARTS_EIA_PKL
+            pickle_plant_parts_distinct(
+                plant_parts_eia=plant_parts_eia, file_path=distinct_file_path
+            )
+        # more efficient memory use for CI
+        if pickle_train_connections:
+            prep_train_connections(plant_parts_eia).to_pickle(
+                pudl_rmi.CONNECTED_TRAIN_PKL
+            )
+
         return plant_parts_eia
 
+    def plant_parts_eia_distinct(self, clobber=False, clobber_ppe=False):
+        """Get the EIA plant_parts with only the unique granularities."""
+        # make this a package variable
+        file_path = pudl_rmi.DISTINCT_PLANT_PARTS_EIA_PKL
+        check_is_file_or_not_exists(file_path)
+        clobber_any = any([clobber, clobber_ppe])
+        if not file_path.exists() or clobber_any:
+            logger.info(
+                f"Distinct EIA plant-parts not found at {file_path}. Generating a new "
+                "distinct dataframe."
+            )
+            # this is weird but the ppe needs to be deleted to free memory
+            ppe = self.plant_parts_eia(  # noqa: F841
+                clobber=clobber_ppe, pickle_distinct=True
+            )
+            del ppe
```

Comment:
Yes? Honestly not sure if assigning the dataframe to `ppe` so that it can be explicitly deleted makes it be cleaned up faster than never saving a reference to it in the first place. This is probably silly and doesn't save memory so I'll try taking it out and just doing `_ = self.plant_parts_eia()`.

Reply:
We're less memory constrained now so I took out this weirdness.